media: i2c: imx477: Add vsync trigger_mode parameter

neocortex-vision opened the pull request:
trigger_mode == 0 (default) => no effect / no registers written
trigger_mode == 1           => source
trigger_mode == 2           => sink

This can be set e.g. in /boot/cmdline.txt as imx477.trigger_mode=N
Setting it to 0 or not setting it at all has the same effect.

Tested on HQ cameras (XVS <-> XVS, GND <-> GND) with libcamera.

Credits to 6by9.

6by9 commented on drivers/media/i2c/imx477.c:
```diff
@@ -25,6 +25,10 @@ static int dpc_enable = 1;
 module_param(dpc_enable, int, 0644);
 MODULE_PARM_DESC(dpc_enable, "Enable on-sensor DPC");
 
+static int trigger_mode = -1;
```
It was fine using a value of 0, but a static variable is automatically zeroed and so didn't need it explicitly.

```static int trigger_mode;```
would have been fine.

6by9 reviewed (commented):
Thanks for the update.

Could you also squash your commits together. We apply them exactly as presented in the PR, so don't need your development history.

neocortex-vision committed:
media: i2c: imx477: Add vsync trigger_mode parameter

trigger_mode == 0 (default) => no effect / no registers written
trigger_mode == 1           => source
trigger_mode == 2           => sink

This can be set e.g. in /boot/cmdline.txt as imx477.trigger_mode=N

Signed-off-by: Jonas Jacob <jonas.jacob@neocortexvision.com>
